cmd/listd/handlers/handlers_test.go

```diff
@@ -32,26 +33,34 @@ var ts testSuite
 // TestMain handles the setup of the testSuite, runs all of the unit tests within
 // the handlers package, and cleans up afterward.
 func TestMain(m *testing.M) {
-	var mainErr error
+	var (
+		err      error
+		dbc      *sqlx.DB
+		exitCode = 1
+	)
+
 	defer func() {
```

Comment:
You could move this after the db connection is established. That would eliminate the extra condition.

Reply:
You can also have multiple defer statements as needed. Which is also clean way of doing it.

Reply:
i would still need the dbc condition and the nil check even if i moved it after the connection is established because if the connection errors there is still a chance dbc could be nil

the order of the if statements were malformed though